apps/wallet/src/background/keyring/VaultStorage.test.ts

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, expect, it, vi } from 'vitest';
+import Browser from 'webextension-polyfill';
+
+import { VaultStorage } from './VaultStorage';
+import { SESSION_STORAGE } from './storage-utils';
+import * as Bip39 from '_shared/utils/bip39';
+import {
+    testEd25519Serialized,
+    testVault,
+    testVault1,
+} from '_src/test-utils/vault';
+
+describe('VaultStorage', () => {
+    describe('create', () => {
+        it('throws if already initialized', async () => {
+            const vaultStorage = new VaultStorage();
+            vi.spyOn(Browser.storage.local, 'get').mockResolvedValue({
+                vault: {},
+            });
+            await expect(vaultStorage.create('12345')).rejects.toThrow(
+                'Mnemonic already exists, creating a new one will override it. Clear the existing one first.'
+            );
+        });
+
+        it('uses random entropy when not provided and creates the vault', async () => {
```

Comment:
I don't like how much this is testing internal implementation details. I don't think it's valuable to test the internal workings, and we should instead just focus on testing the observable behavior (i.e. there is something in the vault storage we can get after initializing).